src/tests/Microsoft.PowerFx.Interpreter.Tests/SetFunctionTests.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.Tests;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    public class SetFunctionTests : PowerFxTest
+    {
+        private readonly ParserOptions _opts = new ParserOptions { AllowsSideEffects = true };
+
+        // $$$ Trigger a recalc?
+        [Fact]
+        public void TestSetVar()
+        {
+            var config = new PowerFxConfig();
+            config.EnableSetFunction();
+            var engine = new RecalcEngine(config);
+
+            engine.UpdateVariable("x", FormulaValue.New(12));
+
+            var r1 = engine.Eval("x", null, _opts); // 12
+            Assert.Equal(12.0, r1.ToObject());
+
+            var r2 = engine.Eval("Set(x, 15)", null, _opts);
+
+            r1 = engine.Eval("x"); // 15
+            Assert.Equal(15.0, r1.ToObject());
+
+            r1 = engine.GetValue("x");
+            Assert.Equal(15.0, r1.ToObject());
```

Comment:
Any test for
engine.UpdateVariable("x", FormulaValue.New(12));
engine.UpdateVariable("y", FormulaValue.New(12));
Set(y, x *2)
y should be 24

Also probably a test where value is a async task.

Reply:
I'll add Async test to ParallelAsync.txt